quixstreams/app.py

```diff
@@ -4,6 +4,7 @@
 import os
 import signal
 import warnings
+import threading
```

Comment:
Question WRT threading (I ask this with lack of experience besides some googling):

Is it definitive that threading is the best approach here vs multiprocessing? I'm guessing we're using threading due to shared resources like the context manager and producer flushing with checkpointing (though I'm sure with the right setup we could use multiprocessing)? 

Wasn't sure if threading would hit limits quicker, as at some point I imagine we will no longer necc be I/O bound if we have enough sources running at once (maybe even just one?). I feel like multiple factors might be at play there though (including hardware related, not sure how fractional CPU in things like Kubernetes affects threading and multiprocessing)

Reply:
We could leverage multiprocessing for sources. It probably even make more sense than threading due to python constraints. We actually don't share too much data from the main thread to the source thread, especially since we are using another producer so it's be worth looking into it. 

It shouldn't be too hard to go from one to the other. Mostly, as you mentioned, handling the checkpointing.

Reply:
swapped to multiprocessing